microdf/generic.py

```diff
@@ -88,13 +87,125 @@ def quantile(self, quantiles):
         return np.interp(quantiles, weighted_quantiles, values)
 
     @handles_zero_weights
-    def median(self):
+    def median(self) -> float:
         """Calculates the weighted median of the MicroSeries.
 
         :returns: The weighted median of a DataFrame's column.
+        :rtype: float
         """
         return self.quantile(0.5)
 
+    def gini(self, negatives: str = None) -> float:
+        """Calculates Gini index.
+
+        :param negatives: An optional string indicating how to treat negative
+            values of x:
+            'zero' replaces negative values with zeroes.
+            'shift' subtracts the minimum value from all values of x,
+            when this minimum is negative. That is, it adds the absolute
+            minimum value.
+            Defaults to None, which leaves negative values as they are.
+        :type q: str
+        :returns: Gini index.
+        :rtype: float
+        """
+        x = np.array(self).astype("float")
+        if negatives == "zero":
+            x[x < 0] = 0
+        if negatives == "shift" and np.amin(x) < 0:
+            x -= np.amin(x)
+        if self.weights is not None:  # Check if it's not np.ones instead?
+            sorted_indices = np.argsort(self.weights)
```

Comment:
```suggestion
            sorted_indices = np.argsort(self)
```
Fixes the gini test for me.